src/pages/discover.js

```diff
@@ -17,83 +17,61 @@ import { H1, P } from '../components/Text';
 import LoadingGrid from '../components/LoadingGrid';
 import Pagination from '../components/Pagination';
 
-const fetchCollectives = lifecycle({
-  async componentDidMount() {
-    const { query } = this.props.router;
-    const params = {
-      offset: query.offset || 0,
-      show: query.show || 'all',
-      sort: query.sort || 'most popular',
-    };
+const _transformData = collective => ({
+  ...collective,
+  stats: {
+    backers: {
+      all: collective.backersCount,
+    },
+    yearlyBudget: collective.yearlyIncome,
+  },
+  type: 'COLLECTIVE',
+});
+
+function useCollectives(query) {
+  const [responseData, setResponseData] = useState({});
+  const params = {
+    offset: query.offset || 0,
+    show: query.show || 'all',
+    sort: query.sort || 'most popular',
+  };
 
+  const fetchDiscoverData = async () => {
     try {
       const endpoints = [`/discover?${queryString(params)}`, '/groups/tags'];
       const [data, tags] = await Promise.all(endpoints.map(e => fetch(`${getBaseApiUrl()}${e}`).then(r => r.json())));
-      this.setState({
+      setResponseData({
         ...pick(data, ['offset', 'total']),
         ...params,
         tags,
-        collectives: data.collectives.map(this.transformData),
-        router: this.props.router,
+        collectives: data.collectives.map(_transformData),
       });
     } catch (error) {
-      this.setState({ error });
+      console.log(error);
```

Comment:
This used to be a setState, now it's a console log?

Reply:
The problem is `error`  state is not used anywhere in the component, making it an unused variable in react hook.